client/src/components/autosave/index.js

```diff
@@ -38,8 +38,8 @@ class Autosave extends React.Component {
   }
 
   tick = () => {
-    const { dispatch, saveInProgress } = this.props;
-    if (this.needToSave() && !saveInProgress) {
```

Comment:
limit the knowledge and responsibility of `tick()` to only checking `this.needToSave()`, since `exponentialBackoffFetch()` already knows to return inflight promise

Additionally, we rely on `tick()` to invoke `exponentialBackoffFetch()` to cancel timeout if `hash` changes